### PR TITLE
Support icmp traffic

### PR DESCRIPTION
**Pull request by mielverkerken**
Add support for icmp traffic. Might be interesting to finish icmp flows on reply after requst.

### CODE REVIEW

**Review by Str-Gen — APPROVED** [review]

File: rustiflow/src/main.rs [hunk 1]
```diff
@@ -1139,6 +1134,22 @@ fn extract_ipv4_features(ipv4_packet: &Ipv4Packet) -> Option<BasicFeaturesIpv4>
...
+            let icmp_type = icmp_packet.get_icmp_type();
+            let icmp_code = icmp_packet.get_icmp_code();
+            source_port = 0; // ICMPv6 does not have ports
+            destination_port = 0;
+            data_length = icmp_packet.payload().len() as u16;
+            header_length = 8; // ICMP header length
+            length = ipv4_packet.get_total_length();
+
+            // Log the ICMP type and code, you can handle specific types/codes if needed
+            println!("ICMP Type: {:?}, Code: {:?}", icmp_type, icmp_code);
```
Comment (matissecallewaert) [inline rustiflow/src/main.rs:1149]: Is this temporary for testing :test_tube:  ?

**Review by matissecallewaert — COMMENTED** [review]
For this to work in real-time mode, the eBPF need adapting. I also changed the base of this PR for better reviewing :smile:

**Review by Str-Gen — APPROVED** [review]

**Review by Str-Gen — APPROVED** [review]

**Review by Str-Gen — APPROVED** [review]

**Review by Str-Gen — APPROVED** [review]

File: rustiflow/src/main.rs [hunk 2]
```diff
@@ -1137,8 +1137,8 @@ fn extract_ipv4_features(ipv4_packet: &Ipv4Packet) -> Option<BasicFeaturesIpv4>
     } else if protocol == IpNextHeaderProtocols::Icmp {
         if let Some(icmp_packet) = IcmpPacket::new(ipv4_packet.payload()) {
             // For ICMP, we will extract the type and code, along with data length
-            let icmp_type = icmp_packet.get_icmp_type();
-            let icmp_code = icmp_packet.get_icmp_code();
+            // let icmp_type = icmp_packet.get_icmp_type();
+            // let icmp_code = icmp_packet.get_icmp_code();
             source_port = 0; // ICMPv6 does not have ports
             destination_port = 0;
             data_length = icmp_packet.payload().len() as u16;
```
[the remark this answers is not in the document]
Comment (matissecallewaert) [reply rustiflow/src/main.rs:1154]: For the realtime, this is possible but it is pointer work to get the header length.

**Review by matissecallewaert — APPROVED** [review]
Looks good to me! For the hard coded header lengths, maybe another pull request that looks into ways to get the header length straight from the packet?

**Pull request merged**